easygcm-lib/src/main/java/eu/inloop/easygcm/ConnectionUtils.java

```diff
@@ -0,0 +1,29 @@
+package eu.inloop.easygcm;
+
+import android.Manifest;
+import android.content.Context;
+import android.content.pm.PackageManager;
+import android.net.ConnectivityManager;
+import android.net.NetworkInfo;
+
+/**
+ * Created by Tomáš Isteník on 22.9.2015.
+ */
+public class ConnectionUtils {
+
+    public static boolean isOnline(Context context) {
+        if (hasAccessNetworkStatePermission(context)) {
+            ConnectivityManager cm = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
+            NetworkInfo netInfo = cm.getActiveNetworkInfo();
+            return (netInfo != null && netInfo.isConnected());
+        }
+        // No permission to access network state was given, return true and hope for the best
```

Comment:
NETWORK_STATE is not a M Runtime permission, so user can't remove it. I guess somebody can remove it from manifest, but that's not common. It's good to keep the permission check, just change a comment a bit to clarify.